packages/contract-call-decoder/src/lib/ContractCallDecoder.ts

```diff
@@ -0,0 +1,191 @@
+  expression: string,
+  abi,
+  transaction: Transaction,
+  provider?
+): Promise<string> {
+  return await radspec(expression, abi, transaction, provider);
+};
+
+export const findSelectorAndAbiItemFromSignatureHash = (
+  functionSignatureHash,
+  abi
+) => {
+  try {
+    const interf = new Interface(abi);
+    const selector = Object.keys(interf.functions).find((selector) => {
+      return interf.getSighash(selector) === functionSignatureHash;
+    });
+    // TODO: handle error
+    return {
+      selector,
+      abi: interf.functions[selector],
+    };
+  } catch (e) {
+    return false;
+  }
+};
+
+type DecodedParam =
+  | unknown
+  | {
+      readonly [index: string]: unknown;
+    };
+
+type DecodedContractCall = {
+  readonly contract: {
+    readonly author: string;
+    readonly title: string;
+    readonly details: string;
+    readonly custom: {
+      readonly [index: string]: string;
+    };
+  };
+  readonly method: {
+    readonly selector: string;
+    readonly details: string;
+    readonly abi: FunctionFragment;
+    readonly returns: string;
+    readonly notice: string;
+    readonly decodedParams: readonly DecodedParam[];
+    readonly params: { readonly [index: string]: unknown };
+    readonly custom: {
+      readonly [index: string]: string;
+    };
+  };
+};
+
+export const decodeContractCall = async (
+  tx: Transaction,
+  options: GetMetadataOptions = {}
+): Promise<DecodedContractCall | false> => {
+  const getMetadataOptions = {
+    ...defaultGetMetadataOptions,
+    ...options,
+    address: tx.to,
+    chainId: options.chainId || tx.chainId,
+  };
+  const metadata = await getMetadataFromAddress(getMetadataOptions);
+
+  const functionSignatureHash = tx.data.slice(0, 10);
+
+  const selectorAndAbi = findSelectorAndAbiItemFromSignatureHash(
+    functionSignatureHash,
+    metadata.output.abi
+  );
+  if (!selectorAndAbi) {
+    return false;
+  }
+  const { selector, abi } = selectorAndAbi;
+
+  // eslint-disable-next-line functional/no-let
+  let radspecEvaluatedNotice;
+  if (metadata.output?.userdoc?.methods[selector]?.notice) {
+    radspecEvaluatedNotice = await evaluate(
+      metadata.output.userdoc.methods[selector].notice,
+      metadata.output.abi,
+      tx,
+      getMetadataOptions.rpcProvider
+    );
+  }
+
+  const iface = new Interface(metadata.output.abi);
+  const decodedParams = iface
+    .decodeFunctionData(selector, tx.data)
+    .map((param) => {
+      return getValueFromDecodedFunctionData(param);
+    });
+
+  const devdoc = metadata.output.devdoc;
+
+  const customFieldsContract = extractCustomFields(devdoc);
+  const customFieldsMethod = extractCustomFields(devdoc.methods[selector]);
+
+  return {
+    contract: {
+      author: devdoc.author,
+      title: devdoc.title,
+      details: devdoc.details,
+      custom: customFieldsContract,
+    },
+    method: {
+      selector,
+      abi: abi,
+      details: devdoc.methods[selector]?.details,
```

Comment:
Are we not evaluating the method devdoc?

Reply:
we are not, right now radspec is not built to evaluate it, we should edit the lirbary, but I don't know if it is worth it

Reply:
What is the difference between evaluating a `notice` string and a `details` string?

Reply:
notice is in userdoc, dev becomes details in devdoc

@notice | Explain to an end user what this does | contract, library, interface, function, public state variable, event
-- | -- | --
@dev | Explain to a developer any extra details | contract, library, interface, function, state variable, event

Reply:
I superficially looked at the code last time, radspec acutally supports decoding other strings.